### PR TITLE
[Bugfix] Parsing Error for Student Questions leading to Bad Request

### LINKED ISSUE #3350: Students and Tutors cannot ask questions in the Question and Answer section
### Describe the bug

In our course Security Engineering, several students and myself have tried to ask questions via the *Questions and Answers* dialog, but the system returns a `Bad Request` error.

#### To Reproduce

1. Write text in text box.
2. Click on Ask question.
3. Banner with `Bad Request`

#### Expected behavior

... 

#### Screenshots

image

### Environment

Chrome/Brave... 

#### Output in Web Console

<details><pre>
zone.js:3382 POST https://artemis.ase.in.tum.de/api/courses/130/student-questions 400
</pre>
</details>

#### Response

```json
{
  "type": "https://www.jhipster.tech/problem/problem-with-message",
  "title": "Bad Request",
  "status": 400,
  "detail": "JSON parse error: Cannot invoke \"com.fasterxml.jackson.databind.JsonDeserializer.getObjectIdReader()\" because \"valueDes\" is null; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Cannot invoke \"com.fasterxml.jackson.databind.JsonDeserializer.getObjectIdReader()\" because \"valueDes\" is null (through reference chain: de.tum.in.www1.artemis.domain.StudentQuestion[\"author\"]->de.tum.in.www1.artemis.domain.User[\"authorities\"])",
  "path": "/api/courses/130/student-questions",
  "message": "error.http.400"
}
```

### PR DESCRIPTION
**Pull request by lschlesinger**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de and locally -> Bug seems to be only in production!

### Motivation and Context
Due to a parsing error, no questions can be posted in exercises and lecture Q&A; Server returns Bad Request as described in the related issue.
Fixes #3350 

### Description

Changes were made for `StudentQuestion`s and `StudentQuestionAnswer`s:
The field author (incl. `authorities` and `guidedTourSettings` which seems to be a problem for the Jackson parser) is not deserialized anymore, but set at the server to the current user right before saving the `StudentQuestion` or `StudentQuestionAnswer`.
Also an issue in `StudentQuestionAnswerResource` was fixed, as the user input in a requestBody of a PUT request was directly written into the database. Now, we allow changes (via PUT) only for the values `answerText`, `verified`, `tutorApproved`.

### Steps for Testing
As the bug was only prevalent in production, it is most important to check whether no regressions are introduced by these changes, i.e. everything should work as before posting/editing with different authorities.

1. Login as student
2. Visit an exercise or lecture page
3. Post questions and answers -> no bad request errors in console or browser should appear
4. Try to edit your questions and answers -> no bad request errors in console or browser should appear, also nothing else should change (such as author, date)
5. Try to edit questions/answers from others which should not be possible
6. Leave on question to be edited (in text or by approval) from an instructor of that course
7. Login as instructor
8. Navigate to a question from a student
9. Edit the text of a student's question -> no bad request errors in console or client should appear, check if the author name changes, after edit -> it should not!
10. Approve an unapproved question > no bad request errors in console or client should appear, check if the author name changes, after edit -> it should not!

### CODE REVIEW

**Review by tobiasjungmann — APPROVED** [review]
Tested on Ts1. Everything worked for me as expected.

File: src/main/webapp/app/overview/student-questions/student-questions.component.ts [hunk 1]
```diff
@@ -122,7 +122,6 @@ export class StudentQuestionsComponent implements OnInit, AfterViewInit {
     addQuestion(): void {
         const studentQuestion = new StudentQuestion();
         studentQuestion.questionText = this.studentQuestionText;
-        studentQuestion.author = this.currentUser;
```
Comment (krusche) [inline src/main/webapp/app/overview/student-questions/student-questions.component.ts:125]: would you need to remove this in the answer on the client as well?

File: src/main/java/de/tum/in/www1/artemis/domain/StudentQuestionAnswer.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 import org.hibernate.annotations.Cache;
 import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.springframework.cloud.cloudfoundry.com.fasterxml.jackson.annotation.JsonProperty;
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/StudentQuestionAnswer.java:9]: I don't think this is the correct import

File: src/main/java/de/tum/in/www1/artemis/domain/StudentQuestion.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 import org.hibernate.annotations.Cache;
 import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.springframework.cloud.cloudfoundry.com.fasterxml.jackson.annotation.JsonProperty;
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/StudentQuestion.java:12]: I don't think this is the correct import

**Review by krusche — CHANGES_REQUESTED** [review]
see my inline feedback

File: src/main/webapp/app/overview/student-questions/student-questions.component.ts (hunk 1 above)
Comment (lschlesinger) [reply src/main/webapp/app/overview/student-questions/student-questions.component.ts:125]: What should be improved is the consistency, that is correct - but in fact, no matter if `studentQuestion.author` and `studentQuestionAnswer.author` are set in the client, as they are not deserialized, they will be effectively set at the server side anyways. I suggest removing this statement in both, `student-questions.component.ts` (already removed) and `student-questions-row.component.ts` where it is currently set, do you agree?

File: src/main/java/de/tum/in/www1/artemis/domain/StudentQuestionAnswer.java (hunk 2 above)
Comment (lschlesinger) [reply src/main/java/de/tum/in/www1/artemis/domain/StudentQuestionAnswer.java]: resolved with d6b9cc6

File: src/main/java/de/tum/in/www1/artemis/domain/StudentQuestion.java (hunk 3 above)
Comment (lschlesinger) [reply src/main/java/de/tum/in/www1/artemis/domain/StudentQuestion.java]: resolved with d6b9cc6

File: src/main/webapp/app/overview/student-questions/student-questions.component.ts (hunk 1 above)
Comment (lschlesinger) [reply src/main/webapp/app/overview/student-questions/student-questions.component.ts:125]: incorporated in d6b9cc6

**Review by MDK5 — APPROVED** [review]
Tested on ts3, no errors appeared.
It might be worth mentioning student questions and answers need to be enabled in the course settings before testing.
Upvoting answers is not possible, but I assume thats a typo in the testing steps.
Code changes look good to me.

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Worked for me as expected.

**Review by katjanakosic — APPROVED** [review]
Worked as expected on TS2

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 1. Everything worked as expected.

File: src/main/webapp/app/overview/student-questions/student-questions.component.ts (hunk 1 above)
Comment (krusche) [reply src/main/webapp/app/overview/student-questions/student-questions.component.ts:125]: thanks

Comment (krusche) [reply src/main/webapp/app/overview/student-questions/student-questions.component.ts:125]: as far as I can see `this.currentUser` is not used anymore and could be removed in this file including the assignment in line 48. I suggest to do that in a follow-up PR

**Review by krusche — APPROVED** [review]

**Pull request merged**